common/src/test/java/com/microsoft/identity/common/ropc/PasswordGrantTest.java

```diff
@@ -0,0 +1,147 @@
+    private static final String CLIENT_ID = "4b0db8c2-9f26-4417-8bde-3f0e3656f8e0";
+
+    private Scenario getDefaultTestScenario() {
+        TestConfigurationQuery query = new TestConfigurationQuery();
+        query.userType = "Member";
+        query.isFederated = false;
+        query.federationProvider = "ADFSv4";
+
+        Scenario scenario = Scenario.GetScenario(query);
+        return scenario;
+    }
+
+    private boolean isEmpty(String[] arr) {
+        return arr == null || arr.length == 0;
+    }
+
+    private String convertScopesArrayToString(final String[] scopes) {
+        if (isEmpty(scopes)) {
+            return null;
+        }
+        final Set<String> scopesInSet = new HashSet<>(Arrays.asList(scopes));
+        return StringUtil.convertSetToString(scopesInSet, " ");
+    }
+
+    private TokenRequest createTokenRequest(String[] scopes, String username, String password) {
+        String scope = convertScopesArrayToString(scopes);
+
+        TokenRequest tokenRequest = new MicrosoftStsTokenRequest();
+        tokenRequest.setClientId(CLIENT_ID);
+        tokenRequest.setScope(scope);
+        tokenRequest.setUsername(username);
+        tokenRequest.setPassword(password);
+        tokenRequest.setGrantType(TokenRequest.GrantTypes.PASSWORD);
+
+        return tokenRequest;
+    }
+
+    @Test
+    public void canPerformROPC() throws IOException {
+
+        Scenario scenario = getDefaultTestScenario();
+        Credential credential = scenario.getCredential();
+
+        AADTestAuthority aadTestAuthority = new AADTestAuthority();
+        OAuth2Strategy testStrategy = aadTestAuthority.createOAuth2Strategy();
+
+        TokenRequest tokenRequest = createTokenRequest(SCOPES, credential.userName, credential.password);
+
+        try {
+            final TokenResult tokenResult = testStrategy.requestToken(tokenRequest);
+
+            assertEquals(true, tokenResult.getSuccess());
+        } catch (ClientException exception) {
+            fail("Unexpected exception.");
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void requestFailsIfUsernameNotProvided() throws IOException {
+
+        Scenario scenario = getDefaultTestScenario();
+        Credential credential = scenario.getCredential();
+
+        AADTestAuthority aadTestAuthority = new AADTestAuthority();
+        OAuth2Strategy testStrategy = aadTestAuthority.createOAuth2Strategy();
+
+        TokenRequest tokenRequest = createTokenRequest(SCOPES, null, credential.password);
+
+        try {
+            final TokenResult tokenResult = testStrategy.requestToken(tokenRequest);
+
+            assertEquals(true, tokenResult.getSuccess());
+        } catch (ClientException exception) {
+            fail("Unexpected exception.");
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void requestFailsIfPasswordNotProvided() throws IOException {
+
+        Scenario scenario = getDefaultTestScenario();
```

Comment:
nit: add final to variables, here and everywhere.

Reply:
Addressed